### PR TITLE
ENH: DatetimeProperties results seem to be inconsistent since missing milliseconds #49073

### LINKED ISSUE #49073: ENH: `DatetimeProperties` results seem to be inconsistent since missing milliseconds
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
In [28]: import pandas as pd

In [29]: p = pd.Series([123123000123123, 1323123132, 23123123123], dtype='datetime64[ns]')

In [30]: p
Out[30]: 
0   1970-01-02 10:12:03.000123123
1   1970-01-01 00:00:01.323123132
2   1970-01-01 00:00:23.123123123
dtype: datetime64[ns]

In [31]: p.dt.second
Out[31]: 
0     3
1     1
2    23
dtype: int64

In [32]: p.dt.nanosecond
Out[32]: 
0    123
1    132
2    123
dtype: int64

In [33]: p.dt.microsecond
Out[33]: 
0       123
1    323123
2    123123
dtype: int64
```

### Issue Description

When `second` & `nanosecond` are being accessed, the results being returned are consistent with expectations i.e., only the `second` & `nanosecond` components of the times. But when `microsecond` is accessed, a result of combining `millisecond` & `microsecond` are being returned. This doesn't seem to be consistent with the way other properties return their results.

This is also a Feature request to have the `millisecond` property added to `DatetimeProperties`.

### Expected Behavior

```python
In [33]: p.dt.microsecond
Out[33]: 
0    123
1    123
2    123
dtype: int64

In [37]: p.dt.millisecond
Out[37]: 
0      0
1    323
2    123
dtype: int16
```

### Installed Versions

<details>

In [38]: pd.show_versions()
/nvme/0/pgali/envs/cudfdev/lib/python3.9/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")

INSTALLED VERSIONS
------------------
commit           : 87cfe4e38bafe7300a6003a1d18bd80f3f77c763
python           : 3.9.13.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-76-generic
Version          : #86-Ubuntu SMP Fri Jan 17 17:24:28 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.5.0
numpy            : 1.23.3
pytz             : 2022.4
dateutil         : 2.8.2
setuptools       : 65.4.1
pip              : 22.2.2
Cython           : 0.29.32
pytest           : 7.1.3
hypothesis       : 6.56.2
sphinx           : 5.2.3
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.5.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           : 
fastparquet      : None
fsspec           : 2022.8.2
gcsfs            : None
matplotlib       : None
numba            : 0.56.2
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 9.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : 2022.8.2
scipy            : 1.9.1
snappy           : 
sqlalchemy       : 1.4.41
tables           : None
tabulate         : 0.9.0
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by PedroPUCRIO**
- [ X] closes #49073 (Replace xxxx with the GitHub issue number)
- [ X] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ X] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

I have added the millisecond feature to the Datetime function on Pandas. As it was discussed in the issue, it seemed inconsistent. However, I didn't change how the microseconds were being exhibited (with 6 digits instead of 3), not to break the code.

### CODE REVIEW

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -1874,6 +1898,7 @@ def to_julian_date(self) -> npt.NDArray[np.float64]:
                 self.hour
                 + self.minute / 60
                 + self.second / 3600
+                + self.millisecond / 3600 / 10**3
```
Comment (mroeschke) [inline pandas/core/arrays/datetimes.py:1901]: Why was this added? I think `self.microsecond` should already include the milliseconds

**Review by mroeschke — CHANGES_REQUESTED** [review]
Thanks for the start:

1. Could you add tests for this functionality?
2. Could you also add this for `Timestamp` for symmetry?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks for working on this, but as mentioned in 

https://github.com/pandas-dev/pandas/issues/49073#issuecomment-1322213775

 I think I'd be -1 on this

**Comment by MarcoGorelli** [comment]
closing as per https://github.com/pandas-dev/pandas/issues/49073#issuecomment-1323967551 , but thanks for your PR!

**Pull request closed without merging**